src/main/webapp/app/shared/constants/file-extensions.constants.ts

```diff
@@ -44,3 +44,301 @@ export const FILE_EXTENSIONS = [
     'odi',
     'odf',
 ];
+
+/**
+ * The list of file extensions that are readable in a file editor.
+ * Extensions are case-sensitive.
+ * Extensions are the part of the filename after the last dot.
+ * Files without dots are represented with their full filename.
+ */
+export const TEXT_FILE_EXTENSIONS = [
```

Comment:
Also `.gradle` for `build.gradle` and `settings.gradle`?

Reply:
Yes, that probably belongs here. But I didn't intend to add any changes to this list in this PR.